src/main/kotlin/org/web3j/quorum/enclave/Tessera.kt

```diff
@@ -26,14 +26,17 @@ class Tessera(private val service: EnclaveService, private val web3: Quorum) : E
         return web3.ethSendRawPrivateTransaction(payload, privateFor).send()
     }
 
+    // payload is hex encoded bytes
+    // hash returned in response is base64 encoded string
     override fun storeRawRequest(payload: String, from: String, to: List<String>): SendResponse {
         val storeRawRequest = StoreRawRequest(payload, from)
         return service.send(storeRawRequest, "storeraw", SendResponse::class.java)
     }
 
-    override fun receiveRequest(key: String, to: String): ReceiveResponse {
-        val receiveRequest = ReceiveRequest(key, to)
-        return service.send(receiveRequest, "receive", ReceiveResponse::class.java)
+    // key is hex encoded payload string
+    // returned string is hex encoded byte string
+    override fun receiveRequest(key: String, to: String): String {
+        return web3.quorumGetPrivatePayload(key).send().getPrivatePayload()
     }
 
```

Comment:
I think the intention with `web3.quorumGetPrivatePayload(key)` is for a user of `web3j-quorum` to be able to get the private payload from their node. It feels a little strange to be using this elsewhere in our code. That said `sendRawRequest` is looking fairly similar!

Reply:
the Tessera 3rd party api which has storeRaw doesn't have a "receive" endpoint, so instead i'm going thru the quorum node to get the response. though i can keep the return type as ReceiveResponse in Enclave.kt and wrap the tessera call

Reply:
I think originally the receiveRequest was copy-pasted from constellation. When it comes to the tessera 3rd party API receiveRequest is an UnsupportedOperation (and an UnsupportedOperationException should be thrown).

Reply:
tessera receive now throws an error